src/Ens.ts

```diff
@@ -0,0 +1,638 @@
+        networkId: this.network,
+        blockchain: BlockchainType.ETH,
+        ownerAddress: (await this.addr(domain, BlockchainType.ETH)) || '',
+        blockchainProviderUrl: this.url,
+      };
+    }
+
+    return result;
+  }
+
+  async getAddress(
+    domain: string,
+    network: string,
+    token: string,
+  ): Promise<string> {
+    throw new ResolutionError(ResolutionErrorCode.UnsupportedMethod, {
+      methodName: 'getAddress',
+      domain,
+      method: this.serviceName(),
+    });
+  }
+
+  /**
+   * This was done to make automated tests more configurable
+   */
+  private resolverCallToName(
+    resolverContract: EthereumContract,
+    nodeHash: string,
+  ) {
+    return this.callMethod(resolverContract, 'name', [nodeHash]);
+  }
+
+  /**
+   * This was done to make automated tests more configurable
+   */
+  private async reverseRegistrarCallToNode(
+    reverseRegistrarContract: EthereumContract,
+    address: string,
+  ): Promise<string> {
+    return await this.callMethod(reverseRegistrarContract, 'node', [address]);
+  }
+
+  // @see: https://docs.ens.domains/ens-improvement-proposals/ensip-5-text-records#service-keys
+  async twitter(domain: string): Promise<string> {
+    try {
+      return await this.record(domain, 'com.twitter');
+    } catch (err) {
+      throw new ResolutionError(ResolutionErrorCode.RecordNotFound, {
+        domain,
+        method: this.name,
+        methodName: 'twitter',
+        recordName: err.recordName,
+      });
+    }
+  }
+
+  async allRecords(domain: string): Promise<Record<string, string>> {
+    throw new ResolutionError(ResolutionErrorCode.UnsupportedMethod, {
+      domain,
+      methodName: 'allRecords',
+    });
+  }
+
+  protected getCoinType(currencyTicker: string): string {
+    const bip44constants = requireOrFail(
+      'bip44-constants',
+      'bip44-constants',
+      '^8.0.5',
+    );
+    const formatsByCoinType = requireOrFail(
+      '@ensdomains/address-encoder',
+      '@ensdomains/address-encoder',
+      '>= 0.1.x <= 0.2.x',
+    ).formatsByCoinType;
+    const coin = bip44constants.findIndex(
+      (item) =>
+        item[1] === currencyTicker.toUpperCase() ||
+        item[2] === currencyTicker.toUpperCase(),
+    );
+    if (coin < 0 || !formatsByCoinType[coin]) {
+      throw new ResolutionError(ResolutionErrorCode.UnsupportedCurrency, {
+        currencyTicker,
+      });
+    }
+
+    return coin.toString();
+  }
+
+  async addr(
+    domain: string,
+    currencyTicker: string,
+  ): Promise<string | undefined> {
+    const resolver = await this.resolver(domain).catch(
+      (err: ResolutionError) => {
+        if (err.code !== ResolutionErrorCode.UnspecifiedResolver) {
+          throw err;
+        }
+      },
+    );
+
+    const isWrappedDomain = await this.determineIsWrappedDomain(
+      this.namehash(domain),
+    );
+    if (isWrappedDomain) {
+      return await this.getAddressForWrappedDomain(domain);
```

Comment:
wrapped and non-wrapped domains should behave the same right? Not sure why we need this part